### PR TITLE
refactor!: asyncio & postgres

### LINKED ISSUE #5: Migrate to asyncio
Currently being done on the asybc branch. And does what it says: migrates from flask and gevent to FastAPI and asyncio.

### LINKED ISSUE #6: Migrate to PostgreSQL
After long consideration and much deliberation, we're going to be moving from MongoDB to PostgreSQL on our async branch. This, for now, will be using asyncpg + SQLAlchemy, although if performance starts getting dicey in the future we may have to use asyncpg alone.

### LINKED ISSUE #13: Beautify README
Currently our readme is really tiny and only holds some information and a notice to not yet contribute without prior notice.
This shall be removed and a proper readme, with a step-by-step guide, requirements, etc. shall be implanted.

This may only happen *after* the async migration to make rebasing easier and other such.

### PR DESCRIPTION
**Pull request by VincentRPS**
Migrates to asyncio and postgresql (via SQLAlchemy 2.0)

Closes #5, Closes #6, and Closes #13

### CODE REVIEW

**Comment by VincentRPS** [comment]
This PR is currently mostly ready.

There are still a few more things to do of course. Remaking tests will happen within a different PR to remove extra clutter, and a bunch more things. I'm gonna leave this open for people to review for a day or two just in case I missed anything.

File: README.md [hunk 1]
```diff
@@ -1,9 +1,47 @@
-# Derailed
+# Derailed API
 
-**This is the Derailed Pre-Alpha API Repository.**
+The most powerful API of the lands.
```
Comment (EnokiUN) [inline README.md:3]: is it?

File: README.md [hunk 2]
```diff
@@ -1,9 +1,47 @@
...
-***Please* do not make any PRs without telling me or else they won't be accepted.**
+Deploying the Derailed API is quite easy, just fill in the
+required `.env` variables and launch either using
+the `gunicorn` command or just running our development build.
+
+You can also use our docker-compose config,
+although this isn't recommended since it
+could make scaling in the future much more
+of a struggle and increase any pain of running
+our services separately.
```
Comment (EnokiUN) [inline README.md:19]: what.

File: README.md (hunk 1 above)
Comment (VincentRPS) [reply README.md:3]: In the case of Derailed, yes.

File: README.md [hunk 3]
```diff
@@ -1,9 +1,47 @@
...
+
+Events are sent through an asynchronous,
+taskful, and stateless gRPC request to either
+our gRPC User or gRPC Guild service.
+
+That service then handles it from there.
+
+## Speedups
+
+At every step, we try to use any speedup possible.
```
Comment (EnokiUN) [inline README.md:31]:
```suggestion
At every step, we try to use every speedup possible.
```

File: README.md [hunk 4]
```diff
@@ -1,9 +1,47 @@
...
+We also use uvloop which is a faster implementation
+of the asyncio event loop which **only works on Linux.**
+Don't worry, we don't force it upon Windows users at all,
+and it's only used when launching our Gunicorn server
+which *also* requires Linux, or our Docker image.
+
+## Benchmarks
+
+We have yet to perform benchmarks on Derailed,
+but in the future we may to see any bottlenecks and other such.
```
Comment (EnokiUN) [inline README.md:45]: this line reads badly

File: README.md (hunk 2 above)
Comment (VincentRPS) [reply README.md:19]: Migrating something like Redis, or PostgreSQL from a docker compose environment would be much harder than just running them as separated services in the first place

File: derailed/app.py [hunk 5]
```diff
@@ -1,83 +1,56 @@
-import dotenv
-from gevent import monkey
+"""
```
Comment (EnokiUN) [inline derailed/app.py:1]: do we really need to do the "add license on every file thing"? it's a lot of lines for 0 benefit

**Review by EnokiUN — CHANGES_REQUESTED** [review]

**Comment by EnokiUN** [comment]
I'm not done with this *yet*

File: README.md (hunk 4 above)
Comment (VincentRPS) [reply README.md:45]: How about:

We have yet to perform benchmarks on Derailed.
Although in the future we may to try and itch out any bottlenecks, and other possible failures.

File: derailed/app.py (hunk 5 above)
Comment (VincentRPS) [reply derailed/app.py:1]: I'm a person very conscious of anyone stealing my code or any work I've done. And that makes it the case that I normally always put licenses on anything I can so I can get credit and if code is stolen, it's blatant and not unknown.

File: README.md (hunk 1 above)
Comment (EnokiUN) [reply README.md:3]: "the lands of Derailed"?

File: derailed/app.py (hunk 5 above)
Comment (EnokiUN) [reply derailed/app.py:1]: if I wanted to steal your code I wouldn't leave the license in yk

File: README.md (hunk 2 above)
Comment (EnokiUN) [reply README.md:19]: you can still easily scale it up tho

**Pull request merged**